regex-assembly/942340.ra

```diff
@@ -3,26 +3,155 @@
 
 ##!+ i
 
+##!> define space_or_word \s*[\s\w+]+
+##!> define quotes [\"'`]\s*
+##!> define digits \d+
+##!> define space \s+
+##!> define optional_space \s*
+
+##!> assemble
+  and
+  nand
+  or
+  xor
+  xxor
+  div
+  like
+  between
+  not
+  ##!=< logical_operators
+##!<
+
+
+##!> assemble
+  &
+  <
+  >
+  ^
+  =
+  ->
+  \*
+  /
+  %
+  ==
+  <>
+  <=
+  >=
+  ##!=< sql_operators
+##!<
+
+
 in\s*?\(+\s*?select
-(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)[\s\w+]+regexp\s*?\(
-(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)[\s\w+]+sounds\s+like\s*?[\"'`]
-(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)[\s\w+]+[=\d]+x
-[\"'`]\s*?\d\s*?--
-[\"'`]\s*?\d\s*?#
-[\"'`][\%&<>^=]+\d\s*?=
-[\"'`][\%&<>^=]+\d\s*?or
-[\"'`][\%&<>^=]+\d\s*?xor
-[\"'`][\%&<>^=]+\d\s*?div
-[\"'`][\%&<>^=]+\d\s*?like
-[\"'`][\%&<>^=]+\d\s*?between
-[\"'`][\%&<>^=]+\d\s*?and
-[\"'`]\W+[\w+-]+\s*?=\s*?\d\W+[\"'`]
-[\"'`]\s*?is\s*?\d.+[\"'`]?\w
-[\"'`]\|?[\w-]{3,}[^\w\s.,]+[\"'`]
-[\"'`]\s*?is\s*?[\d.]+\s*?\W.*?[\"'`]
-\bexcept\s+select\b
-\bexcept\s+values\s*?\(
-[\"'`]\s*(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)array\s*\[
-[\"'`]\s*(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)[\w]+\s*!?~
-[\"'`]\s*(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)[\w]+\s+(?:not\s+)?similar\s+to\s+
-[\"'`]\s*(?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)(?:true|false)\b
+
+##!> assemble
+  ##!=> logical_operators
+  ##!> assemble
+    ||
+    &&
+    ##!=>
+    {{space_or_word}}
+    ##!=>
+    regexp\s*?\(
+    sounds\s+like\s*?{{quotes}}
+    [=\d]+x
+  ##!<
+##!<
+
+##! ORIGINAL :up:
+##! (?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)[\s\w+]+regexp\s*?\(
+##! (?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)[\s\w+]+sounds\s+like\s*?[\"'`]
+##! (?i:(?:n?and|x?x?or|div|like|between|not)\s+|(?:\|\||\&\&)\s*)[\s\w+]+[=\d]+x
+##!
+
+{{quotes}}{{optional_space}}{{digits}}{{optional_space}}--
+{{quotes}}{{optional_space}}{{digits}}{{optional_space}}#
+
+##! ORIGINAL: :up:
+##! [\"'`]\s*?\d\s*?--
+##! [\"'`]\s*?\d\s*?#
+##!
+
+##!> assemble
+  {{quotes}}
+  ##!=>
+
+  ##!> assemble
+    ##!=> sql_operators
+    {{space}}
+  ##!<
+
+  ##!> assemble
+    ##!=> logical_operators
+    {{optional_space}}
+  ##!<
+
```

Comment:
I think you're missing a `##!=>` here. Otherwise, everything below will be an alternation to the stiff above, but it should be concatenated to the end.

Reply:
You resolved this comment but didn't add the `##!=>`. Was that on purpose?